src/modules/shared/pages/account/email-notifications/email-notifications-edit.component.ts

```diff
@@ -14,50 +16,57 @@ import { EmailNotificationsTypeEnum, EmailNotificationsPreferencesEnum, Notifica
 })
 export class PageAccountEmailNotificationsEditComponent extends CoreComponent implements OnInit {
 
-  notificationType: EmailNotificationsTypeEnum;
   notificationListLink: string;
 
-  notificationPreferences: { question: string, items: { value: string, label: string }[] };
+  // Flags
+  isAssessmentType: boolean;
+  isQualifyingAccessorRole: boolean;
 
+  currentUserContext: AuthenticationModel['userContext'];
+
+  formPreferencesList: { value: string, label: string, description: string }[] = [];
+  preferencesReponse: EmailNotificationPreferencesDTO = {};
 
   form = new FormGroup({
-    notificationPreference: new UntypedFormControl('', { validators: CustomValidators.required('Choose at least one section') })
+    preferencesEnabled: new FormArray<FormControl<string>>([], { updateOn: 'change' }),
   }, { updateOn: 'blur' });
 
-
   constructor(
-    private activatedRoute: ActivatedRoute,
-    private notificationsService: NotificationsService
+    private notificationsService: NotificationsService,
+    private authenticationStore: AuthenticationStore
   ) {
 
     super();
 
-    this.notificationType = this.activatedRoute.snapshot.params.notificationType;
     this.notificationListLink = `/${this.stores.authentication.userUrlBasePath()}/account/email-notifications`;
 
-    this.notificationPreferences = {
-      question: `How often do you want to get email notifications about ${this.translate(`shared.catalog.innovation.notification_context_types.${this.notificationType}.title.plural`).toLowerCase()}?`,
-      items: Object.values(EmailNotificationsPreferencesEnum).map(key => ({ value: key, label: this.translate(`shared.catalog.innovation.email_notification_preferences.${key}.me`) }))
-    };
+    this.isAssessmentType = this.stores.authentication.isAssessmentType();
+    this.isQualifyingAccessorRole = this.stores.authentication.isQualifyingAccessorRole();
+
+    this.currentUserContext = this.authenticationStore.getUserContextInfo();
 
-    this.setPageTitle('Edit email notifications', { showPage: false });
-    this.setBackLink('Go back', this.notificationListLink)
+    this.setPageTitle('Select the email notifications you want to receive',  { width: '2.thirds', size: 'l' });
+
+    this.setBackLink('Go back')
 
   }
 
   ngOnInit(): void {
 
-    if (!Object.values(EmailNotificationsTypeEnum).includes(this.notificationType)) {
-      this.redirectTo('not-found');
-    }
-
+    this.setPageStatus('LOADING');
 
     this.notificationsService.getEmailNotificationsPreferences().subscribe(response => {
 
-      const emailNotification = response.find(item => item.notificationType === this.notificationType);
-
-      this.form.get('notificationPreference')!.setValue(emailNotification?.preference);
+      Object.entries(response).filter((item) => item[1] === 'YES').map((item) => (this.form.get('preferencesEnabled') as FormArray).push(new FormControl<string>(item[0])));
```

Comment:
In this case you should use `.forEach` instead of `.map`. Since map creates a new array and in this case you just want to traverse the array.

If you want another option using a normal for loop you could do:

```
for(const [category, preference] of Object.entries(response)) {
    if(preference === 'YES') {
        // add to array
    }
}
```